### PR TITLE
Apply Rector suggestions for PHP 8.1

### PR DESCRIPTION
**Pull request by asumaran**
### Changes proposed in this Pull Request:

Applied suggestions based on Rector's static analysis for PHP 8.1. 

This is a continuation of https://github.com/woocommerce/woocommerce/pull/40801

Ref: D128479-code

### How to test the changes in this Pull Request:

- `plugins/woocommerce/includes/admin/class-wc-admin-importers.php`
    - Add this code snippet to your `functions.php` file or use the [[Code Snippets extension](https://pe.wordpress.org/plugins/code-snippets/)](https://pe.wordpress.org/plugins/code-snippets/), in order to decrease the batch size.
        ```bash
        add_filter('woocommerce_product_import_batch_size', function ($size) {
        	return 1;
        });
        ```
    - Create 2 simple products
    - Go to Products > All Products > Export. Click “Generate CV”
    - Go to Products > All Products > Import. And upload the .csv file from the previous step then run the importer.
    - It should complete without issues. (You might encounter this issue #44784, but it's unrelated to this PR.)
- `plugins/woocommerce/includes/admin/class-wc-admin-webhooks.php`
    - Go to WooCommerce > Settings > Advanced > Webhooks
    - Create a webhook.
    - Click “Delete permanently”
    - It should delete the webhook without issues.
- `plugins/woocommerce/includes/admin/helper/class-wc-helper.php`
    - Add the following snippet using the Code Snippets plugin:
        ```bash
        add_filter(
        'pre_option_woocommerce_feature_marketplace_enabled',
        fn() => 'no'
        );
        ```
    - Go to [https://your.site/wp-admin/admin.php?page=wc-addons](https://wc.jurassic.tube/wp-admin/admin.php?page=wc-addons)
    - Go to “My Subscriptions”
    - Click “Connect” to connect using your A8C account.
    - It should show the subscriptions page without issues.
- `plugins/woocommerce/includes/admin/importers/class-wc-tax-rate-importer.php`
    - Go to WooCommerce > Settings. Check “Enable tax rates and calculations” then save changes.
    - Go to WooCommerce > Settings > Tax > Standard rates.
    - Click “Insert row” then “Export CSV”
    - Click “Import CSV” and upload the exported file. It should import taxes without issues.
- `plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php`
    - Go to Products. Having at least on product created, then click “Export”. Then click “Generate CV”
    - Go to Products. Click “Import” then upload the .csv file from the previous step.
    - Run the importer. It should finish without issues. Clicking “View import log” should also work.
    - Notice the product was probably skipped because it existed already. That’s expected.
- `plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php`
    - Go to Marketing > Coupons.
    - Add a new coupon. Click “Usage restriction”. Under “Products” choose an existing product then click “publish”
    - Go to Marketing > Coupons. You should see the table of coupons. The Product ID should be in the “Product IDs” column.
- `plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php`
    - Go to Products > Add New.
    - In the product type dropdown select “Variable product”
    - Click the “Variations” tab. Should be able to see the empty view without issues.
- `plugins/woocommerce/includes/admin/reports/class-wc-report-sales-by-category.php`
    - Create a product. Add a price and assign a category to the product.
    - Create an order. Set status as “Completed”. Add product from previous step as item.
    - Go to WooCommerce > Reports > Sales by Category.
    - Choose the category from the first step. Click “show”
    - Should be able to see the chart without issues.
- `plugins/woocommerce/includes/admin/settings/class-wc-settings-tax.php`
    - The tax tables should be visible, can insert or remove tax rates without issues.
    - The method `save_tax_classes` is not used anymore. Tax are saved via ajax now.
- `plugins/woocommerce/includes/admin/views/html-admin-page-status-report.php`
    - Go to WooCommerce > Status. The System status report should be visible without issues.
- `plugins/woocommerce/includes/class-wc-ajax.php`
    - Go to Products > Categories.
    - Sorting categories using the ≡ button should work as expected.
- `plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php`
    - No need to test changes in this file as they are deprecated hooks that were never called anyway.
- `plugins/woocommerce/includes/class-wc-geo-ip.php`
    - No need to test this as changes in this file are insignificant. Will never break the app.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Apply Rector suggestions for PHP 8.1
#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/abstracts/abstract-wc-data.php [hunk 1]
```diff
@@ -808,7 +809,7 @@ public function set_props( $props, $context = 'set' ) {
 			}
 		}
 
-		return $errors && count( $errors->get_error_codes() ) ? $errors : true;
+		return $errors && is_countable( $errors->get_error_codes() ) && count( $errors->get_error_codes() ) ? $errors : true;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/abstracts/abstract-wc-data.php]: I think we can leave this change out.  We know $errors is either false or a WP_Error

File: plugins/woocommerce/includes/abstracts/abstract-wc-order.php [hunk 2]
```diff
@@ -1093,7 +1093,9 @@ public function add_item( $item ) {
 		if ( $item_id ) {
 			$this->items[ $items_key ][ $item_id ] = $item;
 		} else {
-			$this->items[ $items_key ][ 'new:' . $items_key . count( $this->items[ $items_key ] ) ] = $item;
+			$items = $this->items[ $items_key ];
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/abstracts/abstract-wc-order.php]: This should be okay to leave this change out since `$this->get_items()` should always return an array

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php [hunk 3]
```diff
@@ -433,7 +433,7 @@ public static function get_filters_counts( $subscriptions = null ) {
 		foreach ( $filters as $key => $count ) {
 			$_subs = $subscriptions;
 			self::_filter( $_subs, $key );
-			$filters[ $key ] = count( $_subs );
+			$filters[ $key ] = is_countable( $_subs ) ? count( $_subs ) : 0;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper.php]: This would probably be cleaner to update the above check on line 429:

```
if ( empty( $subscriptions ) ) {
```
to 
```
if ( ! is_array( $subscriptions ) || empty( $subscriptions ) ) {
```

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php [hunk 4]
```diff
@@ -1231,7 +1231,7 @@ public static function get_local_woo_themes() {
 				if ( is_readable( $txt ) ) {
 					$txt = file_get_contents( $txt );
 					$txt = preg_split( '#\s#', $txt );
-					if ( count( $txt ) >= 2 ) {
+					if ( is_countable( $txt ) && count( $txt ) >= 2 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper.php]: Since we access this as an array directly below, can we change the check to `is_array( $txt ) & ...` instead?

File: plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php [hunk 5]
```diff
@@ -80,7 +80,7 @@
 			</thead>
 			<tbody>
 				<?php
-				if ( count( $errors ) ) {
+				if ( is_countable( $errors ) && count( $errors ) ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php]: Can we change this one to `is_array()` since we try to iterate over it.

File: plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php [hunk 6]
```diff
@@ -136,8 +136,8 @@ protected function render_amount_column() {
 	 */
 	protected function render_products_column() {
 		$product_ids = $this->object->get_product_ids();
-
-		if ( count( $product_ids ) > 0 ) {
+		$count = is_countable( $product_ids ) ? count( $product_ids ) : 0;
+		if ( $count > 0 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php]: This is probably cleaner as `is_array( $product_ids ) && count( $porduct_ids ) > 0`

File: plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-downloads.php [hunk 7]
```diff
@@ -45,8 +45,9 @@ public static function output( $post ) {
 				$product      = null;
 				$loop         = 0;
 				$file_counter = 1;
+				$count        = is_countable( $download_permissions ) ? count( $download_permissions ) : 0;
 
-				if ( $download_permissions && count( $download_permissions ) > 0 ) {
+				if ( $download_permissions && $count > 0 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-downloads.php]: These changes look like a false positive.  `$download_permissions` should always be an array.

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php [hunk 8]
```diff
@@ -64,7 +64,7 @@
 
 	<?php
 	if ( wc_tax_enabled() ) :
-		$total_taxes = count( $order_taxes );
+		$total_taxes = is_countable( $order_taxes ) ? count( $order_taxes ) : 0;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php]: Can we just add `is_countable( $order_taxes )` into the if statement and leave the `count()` within the `for()`

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php [hunk 9]
```diff
@@ -16,7 +16,7 @@
 <div id="variable_product_options" class="panel wc-metaboxes-wrapper hidden">
 	<div id="variable_product_options_inner">
 
-		<?php if ( ! count( $variation_attributes ) ) : ?>
+		<?php if ( ! ( is_countable( $variation_attributes ) && count( $variation_attributes ) ) ) : ?>
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php]: I think the logic here is off.

I think this would fit better:

```
( ! isset( $variation_attributes ) || ! is_array( $variation_attributes ) || count( $variation_attributes ) === 0 )
```

File: plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php [hunk 10]
```diff
@@ -54,7 +54,7 @@ private static function get_current_notice() {
 			return;
 		}
 
-		if ( count( $note_ids ) > 1 ) {
+		if ( is_countable( $note_ids ) && count( $note_ids ) > 1 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php]: This is a false positive.  `$note_ids` is always an array.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php [hunk 11]
```diff
@@ -252,7 +252,7 @@ function( $plugin ) use ( $country, $filter_by, $active_plugins ) {
 								}
 							);
 
-							$columns_count = count( $columns );
+							$columns_count = is_countable( $columns ) ? count( $columns ) : 0;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php]: We can safely assume $columns is an array because of the foreach iterations already used above.

File: plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-classes.php [hunk 12]
```diff
@@ -8,6 +8,8 @@
 if ( ! defined( 'ABSPATH' ) ) {
 	exit;
 }
+
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-classes.php]: I think we can safely revert this file.  The context it's used in sets `$shipping_class_columns` as an array.

File: plugins/woocommerce/includes/admin/views/html-admin-page-addons.php [hunk 13]
```diff
@@ -69,13 +69,16 @@ class="marketplace-header__tab-link marketplace-header__tab-link_helper"
 
 	<div class="wrap">
 		<div class="marketplace-content-wrapper">
-			<?php require __DIR__ . '/html-admin-page-addons-category-nav.php'; ?>
-			<?php if ( ! empty( $search ) && ! is_wp_error( $addons ) && 0 === count( $addons ) ) : ?>
+			<?php
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/views/html-admin-page-addons.php]: We should be able to revert all the changes in this file.  We know $addons is either a WP_Error or an array based on the calling context.

File: plugins/woocommerce/includes/admin/views/html-admin-page-reports.php [hunk 14]
```diff
@@ -35,7 +35,8 @@
 		?>
 	</nav>
 	<?php
-	if ( count( $reports[ $current_tab ]['reports'] ) > 1 ) {
+	$curr_tab_reports_count = is_countable( $reports[ $current_tab ]['reports'] ) ? count( $reports[ $current_tab ]['reports'] ) : 0;
+	if ( $curr_tab_reports_count > 1 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/views/html-admin-page-reports.php]: This one should be okay to revert.  The setup properly defines these as arrays.

File: plugins/woocommerce/includes/class-wc-ajax.php [hunk 15]
```diff
@@ -1879,7 +1879,7 @@ public static function json_search_categories_tree() {
 							$terms_map[ $ancestor ]->children = array();
 						}
 						$item_exists = count(
-							array_filter(
+							(array) array_filter(
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-ajax.php]: This shouldn't be necessary.

File: plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php [hunk 16]
```diff
@@ -117,6 +117,9 @@ public function handle_deprecated_hook( $new_hook, $old_hook, $new_callback_args
 	 * @return mixed
 	 */
 	protected function trigger_hook( $old_hook, $new_callback_args ) {
+		$order_id = null;
+		$item_id = null;
+		$item = null;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php]: I think the proper fix here is just to remove the below case statements as they will never actually fire an action:

```
			case 'woocommerce_order_update_coupon':
			case 'woocommerce_order_update_shipping':
			case 'woocommerce_order_update_fee':
			case 'woocommerce_order_update_tax':
```

File: plugins/woocommerce/includes/class-wc-discounts.php [hunk 17]
```diff
@@ -627,7 +627,9 @@ protected function validate_coupon_usage_limit( $coupon ) {
 					'return'      => 'ids',
 				)
 			);
-			if ( count( $recent_pending_orders ) > 0 ) {
+
+			$recent_pending_orders_count = is_countable( $recent_pending_orders ) ? count( $recent_pending_orders ) : 0;
+			if ( $recent_pending_orders_count > 0 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-discounts.php]: For these where we query through `wc_get_orders()`, we can assume we will always have an array returned since we are certain the `paginate` parameter is not being set.

File: plugins/woocommerce/includes/class-wc-discounts.php [hunk 18]
```diff
@@ -742,7 +744,8 @@ protected function validate_coupon_maximum_amount( $coupon ) {
 	 * @return bool
 	 */
 	protected function validate_coupon_product_ids( $coupon ) {
-		if ( count( $coupon->get_product_ids() ) > 0 ) {
+		$product_ids_count = is_countable( $coupon->get_product_ids() ) ? count( $coupon->get_product_ids() ) : 0;
+		if ( $product_ids_count > 0 ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-discounts.php]: Let's revert this change.  It's okay to assume the prop is an array getting returned.  In the case that it isn't it is probably best that a fatal stops execution - the same goes for the rest of the getter calls in this file.

File: plugins/woocommerce/includes/class-wc-form-handler.php [hunk 19]
```diff
@@ -1099,7 +1099,8 @@ public static function process_registration() {
 				$validation_error  = apply_filters( 'woocommerce_process_registration_errors', $validation_error, $username, $password, $email );
 				$validation_errors = $validation_error->get_error_messages();
 
-				if ( 1 === count( $validation_errors ) ) {
+				$validation_errors_count = is_countable( $validation_errors ) ? count( $validation_errors ) : 0;
+				if ( 1 === $validation_errors_count ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-form-handler.php]: This can be reverted, we know WP_Error will always return an array for `-> get_error_messages()`

**Review by prettyboymp — CHANGES_REQUESTED** [review]
@asumaran, thanks for working on this.  I've made some notes on the changes.  A lot of these are false positives that should be fine with PHP8 given the context where they are used.  I think it's better to keep the code clean here rather than add the extra checks.

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php (hunk 3 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper.php]: good call

File: plugins/woocommerce/includes/class-wc-ajax.php (hunk 15 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-ajax.php]: Rector suggested it. I'll remove.

File: plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php (hunk 16 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php]: niche catch!

File: plugins/woocommerce/includes/abstracts/abstract-wc-data.php (hunk 1 above)
Comment (asumaran) [reply plugins/woocommerce/includes/abstracts/abstract-wc-data.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/abstracts/abstract-wc-order.php (hunk 2 above)
Comment (asumaran) [reply plugins/woocommerce/includes/abstracts/abstract-wc-order.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php (hunk 3 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php (hunk 4 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php (hunk 5 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php (hunk 6 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php (hunk 8 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-downloads.php (hunk 7 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-downloads.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php (hunk 9 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php (hunk 10 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php (hunk 11 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-classes.php (hunk 12 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-classes.php]: Done

File: plugins/woocommerce/includes/admin/views/html-admin-page-addons.php (hunk 13 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/views/html-admin-page-addons.php]: Done

File: plugins/woocommerce/includes/admin/views/html-admin-page-reports.php (hunk 14 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/views/html-admin-page-reports.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php (hunk 16 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/class-wc-discounts.php (hunk 17 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-discounts.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/class-wc-discounts.php (hunk 18 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-discounts.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/class-wc-form-handler.php (hunk 19 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-form-handler.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

File: plugins/woocommerce/includes/class-wc-ajax.php (hunk 15 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-ajax.php]: Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c

**Comment by asumaran** [comment]
@prettyboymp E2E tests seem to fail for unrelated reasons. I've tried running them locally and the results are inconsistent.

Edit: 

I'm investigating if failing tests are actually legit

**Comment by asumaran** [comment]
@prettyboymp I've fixed all test and lint issues. I think this is ready for a second look.

File: plugins/woocommerce/includes/abstracts/abstract-wc-data.php [hunk 20]
```diff
@@ -619,6 +619,7 @@ public static function prime_raw_meta_data_cache( $raw_meta_data_collection, $ca
 	 * @param bool $force_read True to force a new DB read (and update cache).
 	 */
 	public function read_meta_data( $force_read = false ) {
+		$cached_meta     = null;
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/abstracts/abstract-wc-data.php]: This change can be left out.  The only way `$cached_meta` wouldn't be set is if `$cache_loaded` is still false.

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php [hunk 21]
```diff
@@ -63,10 +63,10 @@
 	</td>
 
 	<?php
-	if ( wc_tax_enabled() ) :
-		$total_taxes = count( $order_taxes );
+	if ( wc_tax_enabled() && is_countable( $order_taxes ) ) :
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php]: This change is fine to leave out based on the context of where this template is used. If `wc_tax_enabled() === true`, we know `$order_taxes` is an array of `WC_Order_Item_Tax` items.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php [hunk 22]
```diff
@@ -264,7 +264,7 @@ function( $plugin ) use ( $country, $filter_by, $active_plugins ) {
 							// phpcs:ignore
 							echo $external_link_icon;
 							echo '</a>';
-							if ( count( $plugin_suggestions ) ) {
+							if ( count( (array) $plugin_suggestions ) ) {
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php]: Since `$plugin_suggestions` was already run through `array_filter()`, we know the value is an `array`, so we don't need this change.

File: plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php [hunk 23]
```diff
@@ -175,14 +175,6 @@ protected function trigger_hook( $old_hook, $new_callback_args ) {
 					do_action( $old_hook, $order_id, $item_id, $item, $item->get_product() );
 				}
 				break;
-			case 'woocommerce_order_update_coupon':
```
Comment (prettyboymp) [inline plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php:178]: What is this change for?  Is it related to PHP8 updates?

**Review by prettyboymp — DISMISSED** [review]

File: plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php (hunk 23 above)
Comment (asumaran) [reply plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php:178]: It was suggested by you on previous review https://github.com/woocommerce/woocommerce/pull/41253#discussion_r1402041319

File: plugins/woocommerce/includes/abstracts/abstract-wc-data.php (hunk 20 above)
Comment (asumaran) [reply plugins/woocommerce/includes/abstracts/abstract-wc-data.php]: done https://github.com/woocommerce/woocommerce/pull/41253/commits/0ca0be7618a30860ab866666f7fdfd38852be8b4

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php (hunk 21 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/meta-boxes/views/html-order-refund.php]: done https://github.com/woocommerce/woocommerce/pull/41253/commits/ef5dee203599dbda2131800c53bc6bc363580173 and https://github.com/woocommerce/woocommerce/pull/41253/commits/15e0d3c80b3f2cbcf58d4f249b6dd1e74a9efbea

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php (hunk 22 above)
Comment (asumaran) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php]: done https://github.com/woocommerce/woocommerce/pull/41253/commits/7daccaa3725745371a813f88461fc079f33398b8

**Review by prettyboymp — COMMENTED** [review]
Changes look good. May need to change the changelog entry though.

**Comment by coreymckrill** [comment]
Looks like CI is stuck. I'm going to close and re-open this to see if that fixes it.

**Review by coreymckrill — APPROVED** [review]
👍 Looks good to me. `is_countable` is available starting in PHP 7.3, so meets WC's minimum PHP version requirement of 7.4.

**Comment by asumaran** [comment]
@coreymckrill can we merge this PR?

**Comment by coreymckrill** [comment]
Looks like we still need a new approval from @prettyboymp since there were requested changes.

**Comment by asumaran** [comment]
@coreymckrill, @prettyboymp can you review again? I've already changed the changelog entry.

**Comment by asumaran** [comment]
> Looks like we still need a new approval from @prettyboymp since there were requested changes.

@coreymckrill can we merge this PR? All suggestions made by @prettyboymp have been addressed.
